cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -244,6 +265,54 @@ public void deleteAllApps(HttpRequest request, HttpResponder responder,
     responder.sendString(status.getCode(), status.getMessage());
   }
 
+  private BodyConsumer deployAppFromArtifact(final Id.Namespace namespace, final String appId) throws IOException {
+
+    return new AbstractBodyConsumer(File.createTempFile(appId, ".json", tmpDir)) {
```

Comment:
normally no, but we decided to do this weird thing where the path is the same as the current api which does need a body consumer, so the two need to have the same method signature.